include/swift/AST/Decl.h

```diff
@@ -4788,8 +4788,12 @@ enum class PropertyWrapperSynthesizedPropertyKind {
 /// VarDecl - 'var' and 'let' declarations.
 class VarDecl : public AbstractStorageDecl {
   friend class NamingPatternRequest;
+  friend class GetFieldIndexRequest;
+  friend class ComputeFieldIndicesRequest;
   NamedPattern *NamingPattern = nullptr;
 
+  int fieldIndex = -1;
```

Comment:
This used to be cached as an `unsigned`, are there any issues here in halving the range of values it can store?

It might also be worth moving into the inline bitfield for `VarDecl`, though that would mean you'd likely have to add a private getter and setter for the cached value so the request can access it.